serviceregistration/consul/consul_service_registration.go

```diff
@@ -325,24 +335,29 @@ func (c *ConsulServiceRegistration) NotifySealedStateChange() error {
 	return nil
 }
 
-func (c *ConsulServiceRegistration) checkDuration() time.Duration {
+func (c *serviceRegistration) NotifyInitializedStateChange(isInitialized bool) error {
```

Comment:
Can you add a comment explaining this a little bit more, my assumption is that consul just will not report on initialized state (which is fine) but it's not super clear that's the outcome. A comment could help here.

Reply:
Sure!